torch/quantization/fx/prepare.py

```diff
@@ -417,6 +431,45 @@ def maybe_insert_input_observers_for_node(
     node.args = tuple(new_args)
     node.kwargs = new_kwargs  # type: ignore[assignment]
 
+def maybe_insert_input_equalization_observers_for_node(
+    node: Node,
+    equalization_qconfig: Any,
+    model: torch.nn.Module,
+    modules: Dict[str, torch.nn.Module],
+    graph: Graph,
+    node_name_to_target_dtype: Dict[str, Any],
+) -> None:
+    """
+    If `node` needs to be equalized, find the input/weight observers it needs in
+    `equalization_qconfig`, creates them, and inserts it into `graph`.
+
+    If `node` does not need an equalization observer, returns None.
+    """
+    if equalization_qconfig is None or not node_supports_equalization(node, modules):
+        return
+
+    new_args = []
+    for arg in node.args:
+        if not isinstance(arg, Node) or node_arg_is_bias(node, arg):
+            new_args.append(arg)
+            continue
+
+        is_weight = node_arg_is_weight(node, arg)
+
+        act_eq_process_ctr = equalization_qconfig.weight if is_weight else \
+            equalization_qconfig.input_activation
+
+        new_eq_obs_mod = act_eq_process_ctr()
+        new_eq_obs_node = insert_observer(
+            arg, new_eq_obs_mod, model, modules, graph)
+
+        # set the type, so the next node can read it
+        node_name_to_target_dtype[new_eq_obs_node.name] = node_name_to_target_dtype[arg.name]
```

Comment:
Do we need to set this type for equalization observers? My understanding is that we want the quantization observers to be inserted irrespective of the equalization observers so they shouldn't really be checking the dtype of the eq observer. @vkuzo lmk if this sounds right

Reply:
I tried to set it up so that the quantization observers are inserted irrespective of the equalization observers by setting the dtype of the equalization observers to be the same dtype as the input. In this case, the quantization observers will be essentially checking the input's dtype instead of the equalization observer's dtype.

Reply:
I think it makes sense to set the dtype, we should know the dtype for every node of the graph.